sql/core/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveSessionCatalog.scala

```diff
@@ -312,6 +312,15 @@ class ResolveSessionCatalog(
     case ShowTablesStatement(None, pattern) if isSessionCatalog(currentCatalog) =>
       ShowTablesCommand(None, pattern)
 
+    case ShowTableStatement(namespace, pattern, partitionsSpec) =>
+      val db = namespace match {
+        case Some(namespace) if (namespace.length != 1) =>
+          throw new AnalysisException(
+            s"The database name is not valid: ${namespace.quoted}")
+        case _ => namespace.map(_.head)
+      }
+      ShowTablesCommand(db, Some(pattern), true, partitionsSpec)
```

Comment:
not about this change, but why pattern is optional?

Reply:
ShowTablesCommand groups two different queries
SHOW TABLES ((FROM | IN) multipartIdentifier)?  **(LIKE? pattern=STRING)?**
SHOW TABLE EXTENDED ((FROM | IN) namespace=multipartIdentifier)?  **LIKE pattern=STRING** partitionSpec?  
This reuse is the reason why it is optional.

Reply:
oh I see, there is `showTables`